cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/StorageManager.java

```diff
@@ -93,4 +95,32 @@ public Storage getStorage(StorageType.Type storageType) {
     throw new IllegalArgumentException(
         "No configured storage found for type: " + storageType.getValue());
   }
+
+  /**
+   * Get the storage from the provided path. Iterate over all storages and return the storage which
+   * is configured and its endpoint is the prefix of supplied path. Note: This method assumes both
+   * hdfs and local storage are not configured at the same time since both hdfs and local storage do
+   * not append scheme to the paths. See: https://github.com/linkedin/openhouse/issues/121
+   *
+   * @param path Path that contains the scheme
+   * @return the storage
+   */
+  public Storage getStorageFromPath(String path) {
+    for (Storage storage : storages) {
+      if (storage.isConfigured()) {
+        if (path.startsWith(storage.getClient().getEndpoint())) {
+          log.info("Resolved to {} storage for path {}", storage.getType().toString(), path);
+          return storage;
+        } else if (path.startsWith("/")) {
```

Comment:
can we do `storage.storageType = StorageType.HDFS` and path.startsWith("/")? 

Current condition is weak and might accidentally greenlight incorrect storage

Reply:
done